src/core/public/styles/rendering/_base.scss

```diff
@@ -63,15 +56,24 @@
   }
 }
 
+// Set a body CSS variable for the app container to use - calculates the total
+// height of all fixed headers + the sticky action menu toolbar
+.kbnBody--projectActionMenuVisible {
+  --kbnAppHeadersOffset: calc(var(--kbnHeaderOffset) + var(--kbnProjectHeaderAppActionMenuHeight));
+
+  &.kbnBody--hasHeaderBanner {
+    --kbnAppHeadersOffset: calc(var(--kbnHeaderOffsetWithBanner) + var(--kbnProjectHeaderAppActionMenuHeight));
+  }
+}
+
 .kbnBody--chromeHidden {
   --euiFixedHeadersOffset: 0;
-  &.kbnBody--projectActionMenuVisible {
-    --kbnAppHeadersOffset: 0;
+
+  &.kbnBody--hasHeaderBanner {
+    --euiFixedHeadersOffset: var(--kbnHeaderBannerHeight);
   }
-}
-.kbnBody--chromeHidden.kbnBody--hasHeaderBanner {
-  --euiFixedHeadersOffset: var(--kbnHeaderBannerHeight);
+
   &.kbnBody--projectActionMenuVisible {
-    --kbnAppHeadersOffset: var(--kbnHeaderBannerHeight);
+    --kbnAppHeadersOffset: var(--euiFixedHeadersOffset, 0);
   }
```

Comment:
This I kept in the `.kbnBody--chromeHidden` block because that is a pretty special state that needs to come after all other (in terms of CSS specificity). I also DRY'd it out to simply inherit straight from `--euiFixedHeadersOffset`, which should be set properly from the preceding 6 lines